katello_rapidnode.py

```diff
@@ -292,12 +298,13 @@ def populate_capsules(parent):
     """
     print(colored("Determining all capsules...\n", 'blue', attrs=['bold']))
     capsules = parent_get_capsules()
+    capsule_id = capsule_name = ""
     username, password = get_credentials_parent()
     adminpassword = CONFIG['credentials']['adminpassword']
     print(colored("Populating child capsule with environments...", 'blue',
                   attrs=['bold']))
     for cap in capsules:
-        capsule_id, capsule_name, dummy = cap.split(",")
+        capsule_id, capsule_name = (cap.split(",") + [None]*99)[:2]
```

Comment:
Why not do something like:

```
In [1]: [1][:2]
Out[1]: [1]

In [2]: [1, 2][:2]
Out[2]: [1, 2]

In [3]: [1, 2, 3, 4][:2]
Out[3]: [1, 2]
```

Or using the split:

```
In [6]: 'a'.split(',')[:2]
Out[6]: ['a']

In [7]: 'a, b'.split(',')[:2]
Out[7]: ['a', ' b']

In [8]: 'a, b, c'.split(',')[:2]
Out[8]: ['a', ' b']
```

If the issue you had was the case of returning just one element and you are getting the following:

```
In [9]: a, b = [1]
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-9-47ad9c23b602> in <module>()
----> 1 a, b = [1]

ValueError: need more than 1 value to unpack
```

Then I think you should assert for at least two elements since you are relying on having the capsule_name available. Or if you don't really need the name and want print None instead it could be something like:

```python
capsule_info = cap.split(",")[:2]

if len(capsule_info) < 2:
    capsule_info.append(None)

capsule_id, capsule_name = capsule_info
```

This way it will be more clear what you are trying to accomplish with this piece of code.